workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerBase.cs

```diff
@@ -1,108 +1,158 @@
+        public readonly Type[] Systems = new[]
+        {
+            typeof(SpatialOSReceiveSystem),
+            typeof(SpatialOSSendSystem),
+            typeof(CleanReactiveComponentsSystem)
+        };
 
-            EntityGameObjectLinker = new EntityGameObjectLinker(World, View);
-        }
+        private readonly Dictionary<long, Entity> entityMapping;
 
-        public void Dispose()
+        protected WorkerBase(string workerId, ConnectionConfig config, Vector3 origin, EntityManager entityManager)
         {
-            WorkerRegistry.UnsetWorkerForWorld(this);
-            View.Dispose();
-            World.Dispose();
-        }
+            FindTranslationUnits();
+            // TODO addAllCommandRequestSenders(WorkerEntity, WorkerEntityId);
+            entityMapping = new Dictionary<long, Entity>();
 
-        public virtual void Connect(ConnectionConfig config)
-        {
+            LogDispatcher = new ForwardingDispatcher();
+            this.EntityManager = entityManager;
+            WorkerId = workerId;
             if (config is ReceptionistConfig)
             {
                 if (UseDynamicId)
                 {
-                    WorkerId = GenerateDynamicWorkerId();
+                    WorkerId = $"{GetWorkerType}-{Guid.NewGuid()}";
                 }
 
-                Connection = ConnectionUtility.ConnectToSpatial((ReceptionistConfig) config, GetWorkerType,
+                Connection = ConnectionUtility.ConnectToSpatial((ReceptionistConfig)config, GetWorkerType,
                     WorkerId);
             }
             else if (config is LocatorConfig)
             {
-                Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig) config, GetWorkerType);
+                Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig)config, GetWorkerType);
             }
             else
             {
                 throw new InvalidConfigurationException($"Invalid connection config was provided: '{config}' Only" +
                     "ReceptionistConfig and LocatorConfig are supported.");
             }
+            WorkerEntity = entityManager.CreateEntity(typeof(WorkerEntityTag));
 
             Application.quitting += () =>
             {
                 ConnectionUtility.Disconnect(Connection);
                 Connection = null;
             };
 
-            View.Connect();
+            entityManager.AddComponent(WorkerEntity, typeof(IsConnected));
+            entityManager.AddComponent(WorkerEntity, typeof(OnConnected));
+
+            Origin = origin;
+        }
+
+        public void Dispose()
+        {
+            ConnectionUtility.Disconnect(Connection);
+            EntityManager.DestroyEntity(WorkerEntity);
+            // todo destroy all spatial entities
+            foreach (var translation in TranslationUnits.Values)
+            {
+                translation.Dispose();
+            }
+        }
+
+        internal void CreateEntity(long entityId)
```

Comment:
Would suggest to move the contents (`CreateEntity` and `RemoveEntity`) into the `SpatialOSRecieveSystem` and expose an internal `RegisterEntity` and `DeregisterEntity`

Reply:
will keep in mind for next PR :P